spvenvelope.go

```diff
@@ -0,0 +1,165 @@
+package bc
+
+import (
+	"context"
+	"errors"
+
+	"github.com/libsv/go-bt"
+)
+
+var (
+	// ErrPaymentNotVerified returns if a transaction in the tree provided was missed during verification
+	ErrPaymentNotVerified = errors.New("a tx was missed during validation")
+
+	// ErrTipTxConfirmed returns if the tip transaction is already confirmed
+	ErrTipTxConfirmed = errors.New("tip transaction must be unconfirmed")
+
+	// ErrNoConfirmedTransaction returns if a path from tip to beginning/anchor contains no confirmed transcation
+	ErrNoConfirmedTransaction = errors.New("not confirmed/anchored tx(s) provided")
+
+	// ErrTxIDMismatch returns if they key value pair of a transactions input has a mismatch in txID
+	ErrTxIDMismatch = errors.New("input and proof ID mismatch")
+
+	// ErrNotAllInputsSupplied returns if an unconfirmed transaction in envelope contains inputs which are not
+	// present in the parent envelope
+	ErrNotAllInputsSupplied = errors.New("a tx input missing in parent envelope")
+
+	// ErrNoTxInputsToVerify returns if a transaction has no inputs
+	ErrNoTxInputsToVerify = errors.New("a tx has no inputs to verify")
+
+	// ErrNilInitialPayment returns if a transaction has no inputs
+	ErrNilInitialPayment = errors.New("initial payment cannot be nil")
+
+	// ErrInputRefsOutOfBoundsOutput returns if a transaction has no inputs
+	ErrInputRefsOutOfBoundsOutput = errors.New("tx input index into output is out of bounds")
+)
+
+// SPVEnvelope is a struct which contains all information needed for a transaction to be verified.
+//
+// spec at https://tsc.bitcoinassociation.net/standards/spv-envelope/
+type SPVEnvelope struct {
+	TxID          string                  `json:"txid"`
+	RawTX         string                  `json:"rawTx,omitempty"`
+	Proof         *MerkleProof            `json:"proof,omitempty"`
+	MapiResponses []MapiCallback          `json:"mapiResponses,omitempty"`
+	Parents       map[string]*SPVEnvelope `json:"parents"`
```

Comment:
I am not the most familiar with this syntax, but because some envelopes will have a proof instead of parents, I think this is not always required.

```suggestion
	Parents       map[string]*SPVEnvelope `json:"parents,omitempty"`
```